### PR TITLE
large tests, edit structure, format, and sentences

### PR DESCRIPTION
**Pull request by MattDodsonEnglish**
- Grouping related elements as closely as possible
- Trimming sentences down and removing unneeded verbiage
- Organizing items in a flatter hierarchy 
- Recommends 80% CPU utilization in the monitoring section, since that's recommended in the hardware section.

This PR uses description lists. In fact, this document is essentially a series of lists that describe different considerations for different facets of a load test.

The goal of the lists is to use standardized formatting, with multiple levels of alignment, to create a visual clue that the elements all relate to a single topic.

Ideally, all the scripting stuff would be in one h2, the dd elements would be smaller, and the error messages and options sections would be in definition lists too. But, I decided against it because the ToC goes to only level two, and because our description list component can't handle code blocks. I've made issues about this problems, #998 , #1001 , #1002 .

But I think some aesthetic issues are worth the cleaner organization. And these issues should be fixable.

### CODE REVIEW

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (MattDodsonEnglish) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:347]:
```suggestion
With the limitations mentioned above, we built a [Kubernetes operator](https://github.com/grafana/k6-operator) to distribute the load of a k6 test across a **Kubernetes cluster**. For further instructions, check out [the running distributed k6 tests on Kubernetes tutorial](https://k6.io/blog/running-distributed-tests-on-k8s/).
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:77]:
```suggestion
Keeping this in mind, you should monitor resources while the test runs&mdash;especially when running tests for the first time.
```

"Along with understanding [this] conceptually" sounds too... wordy.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:80]:
```suggestion
The easiest way to monitor the load generator is to open multiple terminals on the machine: one to run k6, and others to monitor the CPU, memory, and network.
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:94]:
```suggestion
As the test runs, these are good indicators to monitor:
```

These are not the only ones.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:108]: I wouldn't phrase it as an outright "mistake". It's certainly an oversight, and it's good practice to think about failure scenarios.

Can we rephrase this in a similar way as it was before?

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:107]:
```suggestion
When running large stress tests, your script shouldn't assume anything about the HTTP response.
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:134]:
```suggestion
To fix this issue, make your checks resilient to any response type.

The following change will handle the exception.
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:208]: Should `abortOnFail` be formatted as code here, same as `discardResponseBodies`? Or is this not possible currently because of #1002?

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:225]:
```suggestion
Refer to this article about [garbage collection](https://javascript.info/garbage-collection) in the V8 runtime. While the JavaScript VM that k6 uses is very different and runs on Go, the general principles apply. Note that memory leaks are still possible in k6 scripts, and, if not fixed, they might exhaust RAM much more quickly.
```

Sounds more natural to me.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:237]: Might be good to expand on this.

```suggestion
: k6 needs a significant amount of memory when uploading files. Every VU is independent and has its own memory, and files will be copied in all VUs that upload them. You can follow [issue #1931](https://github.com/grafana/k6/issues/1931) that aims to improve this situation.
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:248]:
```suggestion
**Tip:** it's often possible to launch "spot instances" of the same hardware for 10-20% of the cost.
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:266]: I prefer "was able to" here. It has more of an "ability" tone of something that happened in the past.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:274]:
```suggestion
This error is similar to `context deadline exceeded`, but in this case, k6 wasn't even able to make an HTTP request. The target system can't establish a TCP connection.
```

Same here about "could not".

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:284]:
```suggestion
Refer to the ["Fine tuning OS" article](/misc/fine-tuning-os/#viewing-limits-configuration).
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:347]:
```suggestion
With the limitations mentioned above, we built a [Kubernetes operator](https://github.com/grafana/k6-operator) to distribute the load of a k6 test across a **Kubernetes cluster**. For further instructions, check out [the tutorial for running distributed k6 tests on Kubernetes](https://k6.io/blog/running-distributed-tests-on-k8s/).
```

**Review by imiric — CHANGES_REQUESTED** [review]
Nice work @MattDodsonEnglish! :clap:

It looks much better now. :+1: And you were right, `DescriptionList` is a good fit, and separates some of the topics nicely.

Just left some minor comments. Some more nitpicky than others :wink:

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:58]: It's a feature of extended markdown: https://www.markdownguide.org/extended-syntax#definition-lists
I don't like the syntax, but it's gained adoption.


The parsers of many SSGs use it, so it'd work in Hugo out of the box. Our Gatsby component has the disadvantage of requiring a `<DescriptionList>` tag, but the advantage of automatically generating `id` attributes, so each term is has an anchor.

I don't think it works it Github though
: does it?

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (MattDodsonEnglish) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:108]:
```suggestion
An oversight of some scripts is to test with only the [happy path](/misc/glossary#happy-path "The default behavior that happens when the system returns no errors") in mind.
```

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:208]: Exactly. I'd prefer to format this and the error headings as titles, but those are the breaks.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:266]: How about "can"? I should've realized the real problem was the needless past tense.

Comment (MattDodsonEnglish) [inline src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:266]:
```suggestion
This happens when k6 can send a request, but the target system doesn't respond in time. The default timeout in k6 is 60 seconds. If your system doesn't produce the response in this time frame, this error will appear.
```

**Comment by MattDodsonEnglish** [comment]
> It looks much better now. +1 And you were right, `DescriptionList` is a good fit, and separates some of the topics nicely.
> 
> Just left some minor comments. Some more nitpicky than others wink

Thanks! These were all great. Really, you did all the work I should have done myself.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:58]: Ah, got it. I just noticed it's not just line breaks, but separate elements.

Then I'd argue we're abusing the purpose of _definition_ lists, just because of how they're currently styled. It's akin to using tables for layout in the old days, and goes against the semantic HTML concept.

There's no reason these couldn't be wrapped in an element just like `<DescriptionList>`, but contain plain `h3`s or whatever level is appropriate, and then the paragraphs could just have a small `padding-left`. It would require trivial amounts of CSS, the markup would be much simpler, and you could just do a double line break as usual to get new paragaphs. And we wouldn't need to fix issues like #1002, at least not right now.

I don't want to block this PR from merging because of this, but I can create an issue if you agree.

Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:58]: I don't think it's abusing a `<dl`>`. It is actually called a _description list_ by MDN. I swear it used to be called definition list and then they broadened the name. I'm not going to dig, but the standard says the lists can be any name-value groups:

```
Name-value groups may be terms and definitions, metadata topics and values, questions and answers, or any other groups of name-value data.
```

In these cases, I think the description list works well semantically, because these all are lists in one distinct category, e.g. "Specific hardware considerations".  In another form, the info could be represented as a table, since their structure is so similar. On the other hand, the grouping of h3s can be a little looser and usually don't have the same parallellism between heading and content.

But, this is very, very pedantic and I'm being a little bit vague and mystical. Still, I don't think it's so important and I don't think we're doing anything that abuses web conventions or harms accessibility. If you think we are, please make an issue.

Comment (imiric) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md:58]: Fair enough, the naming is confusing, but all the usage examples point to this being used for defining or describing terms, for metadata, and generally simpler usage than what we're using them for (#1002 would be something we shouldn't need).

The reason the colon is used is precisely to define the term, and that's not our use case here. On the glossary page it makes perfect sense, but rarely elsewhere, and definitely not just for indentation.

I'd also point out [this note from MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl#notes):

> Do not use this element (nor `<ul>` elements) to merely create indentation on a page. Although it works, this is a bad practice and obscures the meaning of description lists.
>
> To change the indentation of a description term, use the CSS `margin` property.

Which is exactly what we're doing here. It also has accessibility implications for screen readers, etc., and we don't need to complicate things.

So let's merge this as is, but I'll create the issue to change this.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md]: But the sentence is in past tense already, "can" wouldn't be a good fit either.

Why are you against "was able to"?

**Review by imiric — APPROVED** [review]
LGTM, besides https://github.com/grafana/k6-docs/pull/1003#discussion_r1089087450, which we can address later.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md]: The main clause is "This happens", which is present.

> Why are you against "was able to"?

Ha, yes, you've hit the heart of the issue.

"Is able to" longer than "can". That's the entire reason.

 I'm also generally against the past tense because it creates longer constructions and creates a grammatical distance that isn't necessary unless describing real pat tenses.

Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md]: But this is an extremely minor point. I'm fine with you choosing whatever you prefer, and merging when you think it's ready.

**Comment by MattDodsonEnglish** [comment]
Nice, thanks for your great help (and patience :-D), I don't want to merge
and create any conflicts for you in the main PR, so feel free to merge
whenever you think it's ready.

File: src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md
Comment (imiric) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md]: We're really bikeshedding at this point :sweat_smile:, and I don't want to drag this on, but "this happens" is describing an action that happened in the past. I.e. you ran the test, got the error, and then "this happens when ...". You could write it either way, but describing a past action is more relatable to how the user would experience it.

Still, regardless of the tense, even the present "is able to" would be better than "can", since "can" implies that k6 has the _capability_ to do something, which, of course it does, it _can_ make requests; but "able to" implies the _ability_ in that moment, i.e. it's _unable_ to because of this or that reason.

But nevermind, it's fine as is. :smile:

Comment (MattDodsonEnglish) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md]: If you search google for "this {happens|occurs} when", you'll see most sentences are in the present tense. It's because the phrase introduces a general condition. In our sentences, no event _really_ occurred; we just describe the general conditions that create the error, as we understand them presently.

But, it's an interesting point about "can" vs. "is able to". I think in most cases, I'd only need "is unable to" but you've definitely got me questioning my "no-is-able-to" orthodoxy here.

I now realize the most direct way would be to remove all auxiliary verbs:

> This happens when k6 sends a request and the target system doesn't respond in time.

This particular case is not a big deal in the scheme of things, but I do appreciate any discussion that brings more attention to precision and clarity in language.

Comment (imiric) [reply src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md]: Good points :+1: I agree that the simplest form is to remove all auxiliary verbs, as in your example there. Consistency is probably most important, and you have the big picture in mind. So I leave this up to you.

Thanks for the discussion :slightly_smiling_face:

**Pull request merged**
